.github/workflows/e2e_test.yml

```diff
@@ -0,0 +1,33 @@
+name: E2E Tests
```

Comment:
This workflow is missing the `GO_TOOLCHAIN=local` env.

In general I consider e2e tests to be a part of the CI, so I would move them to `ci.yml`.